### DISCUSSION TITLE
ESP Question

### DISCUSSION THREAD

**Discussion by jkms** [gcormier/megadesk#112] [General]
I have a megadesk PCB and it works great (thanks, I love it!).

Also looking at the recent esphome.md, I was very excited to try and hook my megadesk up to home assistant.  However, I also see an `esp` branch with some `megadeskcompanion` notes.

If you're going to build and sell something prettier than the hackjob of me soldering an esp32 to the serial pins on my existing PCB, I'd rather wait for that.

Are you cooking up something worth waiting for? :)

**Comment by gcormier** [comment]
It's definitely not coming anytime soon.

Even if the hardware worked today (it doesn't :) ), there's a few other hurdles

1. The MCU needs to run in realtime to control the LIN bus and the legs.
2. One cannot simply flash ESPHOME most likely, as it would not have the realtime code to handle LIN.
3. It might simply be better to implement a similar model as we have currently, and look for some sort of Serial over WiFi implementation.
4. Arduino+PlatformIO do not yet support the ESP32-C3-MINI.
